test/test_torch.py

```diff
@@ -217,6 +217,22 @@ def fn(t, dim, keepdim=False):
             self.assertEqual(x.ndimension(), fn(x, dim).ndimension())
             self.assertEqual(x.ndimension(), fn(x, dim, keepdim=True).ndimension())
 
+            # check reducing of a singleton dimension
+            dims = [3, 4, 5]
+            singleton_dim = random.randint(0, 2)
+            dims[singleton_dim] = 1
+            x = torch.randn(dims)
+            fn_attr = getattr(torch, fn_name) if fn_name != "norm" else normfn_attr
+
+            def fn(t, dim, keepdim=False):
+                ans = fn_attr(x, dim, keepdim=keepdim)
```

Comment:
same here.  One issue might be that this function (and the above function) take in a "t" but use "x", so you have to redefine.  Can you fix that above and remove this?